xds/src/main/java/io/grpc/xds/ClusterManagerLoadBalancer.java

```diff
@@ -101,23 +102,32 @@ protected Map<Object, ChildLbState> createChildLbMap(ResolvedAddresses resolvedA
    */
   @Override
   public Status acceptResolvedAddresses(ResolvedAddresses resolvedAddresses) {
-    try {
-      resolvingAddresses = true;
-
-      // process resolvedAddresses to update children
-      AcceptResolvedAddrRetVal acceptRetVal =
-          acceptResolvedAddressesInternal(resolvedAddresses);
-      if (!acceptRetVal.status.isOk()) {
-        return acceptRetVal.status;
+    if (lastResolvedAddresses != null) {
+      // Handle deactivated children
+      ClusterManagerConfig config = (ClusterManagerConfig)
+          resolvedAddresses.getLoadBalancingPolicyConfig();
+      ClusterManagerConfig lastConfig = (ClusterManagerConfig)
+          lastResolvedAddresses.getLoadBalancingPolicyConfig();
+      Map<String, PolicySelection> adjChildPolicies = new HashMap<>(config.childPolicies);
+      for (Entry<String, PolicySelection> entry : lastConfig.childPolicies.entrySet()) {
+        ClusterManagerLbState state = (ClusterManagerLbState) getChildLbState(entry.getKey());
+        if (adjChildPolicies.containsKey(entry.getKey())) {
+          if (state.deletionTimer != null) {
+            state.reactivateChild();
+          }
+        } else if (state != null) {
+          adjChildPolicies.put(entry.getKey(), entry.getValue());
+          if (state.deletionTimer == null) {
+            state.deactivateChild();
+          }
+        }
       }
-
-      // Update the picker
-      updateOverallBalancingState();
-
-      return acceptRetVal.status;
-    } finally {
-      resolvingAddresses = false;
+      config = new ClusterManagerConfig(adjChildPolicies);
+      resolvedAddresses =
+          resolvedAddresses.toBuilder().setLoadBalancingPolicyConfig(config).build();
     }
+    lastResolvedAddresses = resolvedAddresses;
+    return super.acceptResolvedAddresses(resolvedAddresses);
```

Comment:
What keeps this from calling `shutdown` on all of the removed addresses in MultiChildLoadBalancer.java:127 `      shutdownRemoved(acceptRetVal.removedChildren);`